marahgh/assignment1/Q2.cpp

```diff
@@ -0,0 +1,107 @@
+
+
+#include <iostream>
+#include <ext/slist>
+
+using namespace std;
+
+
+//----------------------------------------------------------------------------------//
+/* A generic list with basic functionality, constructor, destructor,
+    insertion to the end if the list and removing the last element of the list*/
+template< typename T>
+class List{
+/*Node's list class */
+template <typename S>
+class Node{
+    public:
+    S data;
+    Node<S>* next;
+    public:
+    Node(S data):data(data),next(nullptr){
+    }
+};    
+	private:
+	Node<T>* head;
+	Node<T>* tail;
+	int size;
+	
+	public:
+	
+	List():head(nullptr),tail(nullptr),size(0){}
+	
+	Node<T>* getHead(){

```

Comment:
Returning a Node* lets the user do whatever it wants with it. It's safer if the interface of the list just receives/returns objects of type T